Added listtransactions RPC

DennisAMenace opened the pull request:
Task notes:

A little different from listtransactions (0.20.0 RPC)

- The result does not have all of the data that the (0.20.0 RPC) provides, but I think we can keep adding to this later in more PR's as needed.

dangershony commented:
> label (AKA account in previous versions of BTC Core) was depreciated in 0.14 so I added this parameter for backwards compatibility, but it does nothing.

What backwards compatibility needs that and cant we just fix it there?
I am afriad we will carry this tech debt for a long tie if we put it back

DennisAMenace commented:
> > label (AKA account in previous versions of BTC Core) was depreciated in 0.14 so I added this parameter for backwards compatibility, but it does nothing.
> 
> What backwards compatibility needs that and cant we just fix it there?
> I am afriad we will carry this tech debt for a long tie if we put it back

No problem, Done. I added this in to allow the selection of the account in the last commit.

dangershony commented on src/External/NBitcoin/ChainIndexer.cs:
```diff
@@ -179,7 +179,7 @@ public virtual IEnumerable<ChainedHeader> EnumerateAfter(ChainedHeader block)
 
         /// <summary>
-        /// TODO: Make this internal when the component moves to Stratis.Bitcoin
+        /// TODO: Make this internal when the component moves to Blockcore
```
ah brilliant job

dangershony commented on src/Features/Blockcore.Features.Notifications/AssemblyInfo.cs:
```diff
@@ -0,0 +1,3 @@
+﻿using System.Runtime.CompilerServices;
+
+[assembly: InternalsVisibleTo("Blockcore.Features.Notifications.Tests")]
```
check with @sondreb he solved this on the property files I think

DennisAMenace replied:
Great find, I confirmed with Sondre and resolved this

dangershony commented on src/Tests/Blockcore.IntegrationTests/GlobalSuppressions.cs:
```diff
@@ -4,14 +4,14 @@
-[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "<Pending>", Scope = "member", Target = "~M:Stratis.Bitcoin.IntegrationTests.Miners.ProofOfStakeMintCoinsSpecification.pow_wallet_broadcasts_tx_of_million_coins_and_pos_wallet_receives")]
-[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "<Pending>", Scope = "member", Target = "~M:Stratis.Bitcoin.IntegrationTests.SmartContracts.SmartContractMinerTests.SmartContract_ReferencingInputInSameBlock~System.Threading.Tasks.Task")]
\ No newline at end of file
+[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Style", "IDE1006:Naming Styles", Justification = "<Pending>", Scope = "member", Target = "~M:Blockcore.IntegrationTests.Miners.ProofOfStakeMintCoinsSpecification.a_proof_of_stake_node_with_wallet")]
```
ah good good I cant believe I missed that

dangershony commented:
@DennisAMenace please just fix the conflict and we push this
